panoply/records.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+from typing import TypedDict, Dict, List
+
+from .errors.exceptions import WrongTypeOrValueError
+
+
+class PanoplyMetadata(TypedDict):
+    resource_id: str
+    timestamp: str
+
+
+class PanoplyRecord(TypedDict):
+    data: Dict
+    metadata: PanoplyMetadata
+
+
+class PanoplyRecordGroup(TypedDict):
+    data: List[Dict]
+    metadata: PanoplyMetadata
```

Comment:
@yonivy I added the prefix because some sources already have `Resource` class. But if it is crucial to leave it as `Resource` we will make changes inside the data source.

Reply:
Yes, that's the preferred way. We shouldn't have "messy semantics" here because some sources already named something similarly.